lib/taiko.js

```diff
@@ -2426,75 +2426,63 @@ module.exports.comboBox = (attrValuePairs, ...args) => {
-          if (
-            this.options[i].text === value ||
-            this.options[i].value === value
-          ) {
-            this.selectedIndex = i;
-            found_value = true;
-            let event = new Event('change');
-            this.dispatchEvent(event);
-            break;
-          }
-        }
-        return found_value;
-      }
-      const options = setNavigationOptions({});
-      await doActionAwaitingNavigation(options, async () => {
-        const result = await runtimeHandler.runtimeCallFunctionOn(
-          selectBox,
-          null,
-          { nodeId: nodeId, arg: value },
-        );
-        if (!result.result.value)
-          throw new Error('Option not available in drop down');
-      });
-      descEvent.emit('success', 'Selected ' + (value.index || value));
-    },
-    value: async () => {
-      const nodeId = (await getIfExists(get)())[0];
-      if (!nodeId)
-        throw `${desc(selector, 'label', 'Dropdown')} not found`;
-      return (await evaluate(nodeId, function getvalue() {
-        return this.value;
-      })).value;
-    },
-    text: selectorText(get),
-  };
+    validate();
+    const {selector} = prepareParameters(attrValuePairs, ...args);
+    const get = getElementGetter(selector,
+        async () => await $$xpath(`//select[@id=(//label[contains(string(), ${xpath(selector.label)})]/@for)] | //label[contains(string(), ${xpath(selector.label)})]/select`),
+        '//select');
+    return {
+        get: getIfExists(get),
+        exists: exists(getIfExists(get), descEvent),
+        description: desc(selector, 'label', 'Dropdown'),
+        select: async (value) => {
+            const nodeId = (await getIfExists(get)())[0];
+            if (!nodeId) throw `${desc(selector, 'label', 'DropDown')} not found`;
+            if (defaultConfig.headful) await highlightElemOnAction(nodeId);
+
+            function selectBox(value) {
+                let found_value = false;
+                if(this.options.length > value.index){
+                    this.selectedIndex = value.index;
+                    found_value = true;
+                    let event = new Event('change');
+                    try{
+                        this.dispatchEvent(event);
+                    }catch(e){
+                        return {'message': 'Error occured while dispatching event','stack': e};
+                    }
+                    return found_value;
+                }
+                for (var i = 0; i < this.options.length; i++) {
+                    if (this.options[i].text === value || this.options[i].value === value) {
+                        this.selectedIndex = i;
+                        found_value = true;
+                        let event = new Event('change');
+                        try{
+                            this.dispatchEvent(event);
+                        }catch(e){
+                            return {'message': 'Error occured while dispatching event','stack': e};
```

Comment:
The function being executed on the browser console can not return non primitive values unless [returnByValue](https://github.com/getgauge/taiko/blob/14d7793ce9d59ace9a9ad70fbf9750b56672ede5/lib/handlers/runtimeHandler.js#L44) is set to true.

Reply:
@negiDharmendra Updated the logging structure. Please check it once.